community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringSchemaKey.java

```diff
@@ -0,0 +1,226 @@
+        {
+            throw new IllegalArgumentException( "Tried to create key without value" );
+        }
+        if ( !Values.isTextValue( values[0] ) )
+        {
+            throw new IllegalArgumentException(
+                    "Key layout does only support strings, tried to create key from " + values[0] );
+        }
+        return (TextValue) values[0];
+    }
+
+    @Override
+    public String propertiesAsString()
+    {
+        return asValue().toString();
+    }
+
+    @Override
+    public Value asValue()
+    {
+        return bytes == null ? Values.NO_VALUE : Values.utf8Value( bytes );
+    }
+
+    // TODO perhaps merge these lowest/highest methods into parent
+    @Override
+    public void initAsLowest()
+    {
+        bytes = null;
+        entityId = Long.MIN_VALUE;
+        entityIdIsSpecialTieBreaker = true;
+    }
+
+    @Override
+    public void initAsHighest()
+    {
+        bytes = null;
+        entityId = Long.MAX_VALUE;
+        entityIdIsSpecialTieBreaker = true;
+    }
+
+    private boolean isHighest()
+    {
+        return entityIdIsSpecialTieBreaker && entityId == Long.MAX_VALUE && bytes == null;
+    }
+
+    /**
+     * Compares the value of this key to that of another key.
+     * This method is expected to be called in scenarios where inconsistent reads may happen (and later retried).
+     *
+     * @param other the {@link StringSchemaKey} to compare to.
+     * @return comparison against the {@code other} {@link StringSchemaKey}.
+     */
+    int compareValueTo( StringSchemaKey other )
+    {
+        // TODO cover all cases of bytes == null and special tie breaker and document
+        if ( bytes != other.bytes )
+        {
+            if ( bytes == null )
+            {
+                return isHighest() ? 1 : -1;
+            }
+            if ( other.bytes == null )
+            {
+                return other.isHighest() ? -1 : 1;
+            }
+        }
+        else
+        {
+            return 0;
+        }
+
+        try
+        {
+            // TODO change to not throw
+            return codePointByteArrayCompare( bytes, other.bytes );
+        }
+        catch ( Exception e )
+        {
+            // We can not throw here because we will visit this method inside a pageCursor.shouldRetry() block.
+            // Just return a comparison that at least will be commutative.
+            return byteArrayCompare( bytes, other.bytes );
+        }
+    }
+
+    private static int byteArrayCompare( byte[] a, byte[] b )
+    {
+        assert a != null && b != null : "Null arrays not supported.";
+
+        if ( a == b )
+        {
+            return 0;
+        }
+
+        int length = Math.min( a.length, b.length );
+        for ( int i = 0; i < length; i++ )
+        {
+            int compare = Byte.compare( a[i], b[i] );
+            if ( compare != 0 )
+            {
+                return compare;
+            }
+        }
+
+        if ( a.length < b.length )
```

Comment:
Integer.compare( a.length, b.length )

Reply:
All of the lines below the first comparison can be replaced with only one integer.compare thingy